AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -295,6 +312,8 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
             // mutation?
             switch deleteInput {
             case .withIdentifier, .withIdentifierAndCondition:
+                let syncDeletionsCount = (queryAndDeleteResult.associatedModels.count + queryAndDeleteResult.deletedModels.count)
+                self.log.debug("[CascadeDelete.4] sending \(syncDeletionsCount) delete mutations")
                 syncDeletions(withModels: queryAndDeleteResult.deletedModels,
```

Comment:
`syncDeletions` is called in at least two places, here and in the `case .withFilter(let filter)` just below this. If we move this log line into `syncDeletions` then it will we'll automatically gain this log for both code paths

Reply:
Adding the two numbers together seems like we're losing more information than we originally started with, if we add this to `syncDeletions(withModels:)` call, and don't add the numbers together, the log line will look like this

```
Begin syncing  \(models.count) \(modelName) models for deletion. 
```

For associated models, I can see this is lacking some information for us since it contains a count of all associated models, and we don't know the subset count for each model. I think this is okay for now since the logging context contains a log above this that says how many of each associated model was queried.
```
Begin syncing \(associatedModels.count) associated models for deletion. 
```

Reply:
sure, added that! I thought the combined numbers(associatedModel.count + deletedModel.count) will give clear understanding of the total no.of models that are sent for delete Mutations. Along with your suggestions, now the logs look like:
```
[CascadeDelete.4] sending a total of 3 delete mutations
[CascadeDeleteOperation<Blog>] [CascadeDelete.4] Begin syncing 1 Blog models for deletion
[CascadeDeleteOperation<Blog>] [CascadeDelete.4] Begin syncing 2 associated models for deletion.
[CascadeDeleteOperation<Blog>] syncDeletions(withModels:predicate:associatedModels:syncEngine:completion:) successfully submitted Comment to sync engine MutationEvent(...)
syncDeletions(withModels:predicate:associatedModels:syncEngine:completion:) successfully submitted Post to sync engine MutationEvent(...)
syncDeletions(withModels:predicate:associatedModels:syncEngine:completion:) successfully submitted Blog to sync engine MutationEvent(...)
```

Reply:
sounds good, i see the latest code has been moved to `syncDeletion`, i've added new comments w.r.t those places